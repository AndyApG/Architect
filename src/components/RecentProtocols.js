import React, { Component } from 'react';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { Flipped } from 'react-flip-toolkit';
import PropTypes from 'prop-types';
import ProtocolStack from './ProtocolStack';

class RecentProtocols extends Component {
  static propTypes = {
    recentProtocols: PropTypes.array.isRequired,
    show: PropTypes.bool,
  };

  static defaultProps = {
    show: true,
  };

  renderRecentProtocol = protocol => (
    <div
      key={protocol.filePath}
      className="recent-protocols__protocol"
    >
      <Flipped flipId={protocol.filePath}>
        <ProtocolStack
          protocol={protocol}
        />
      </Flipped>
    </div>
  );

  renderWelcomeText = () => (
    <div className="recent-protocols__welcome">
      <h1>Getting Started</h1>
      <p>
        Welcome to Network Canvas Architect! To get started, use the buttons above to create a new
        interview protocol, or open an existing one from elsewhere. When you return to this screen
        later, recent protocols you have opened will be shown here.
      </p>
    </div>
  );

  renderProtocolList = recentProtocols => (
    <React.Fragment>
      <h3 className="recent-protocols__title" key="heading">Recently Opened Protocols</h3>
      <div className="recent-protocols__wrapper">
        {recentProtocols.map(this.renderRecentProtocol)}
      </div>
    </React.Fragment>
  );

  render() {
    const { show, recentProtocols } = this.props;

    if (!show) { return null; }

    return (
      <div className="recent-protocols">
        { (recentProtocols.length === 0) ?
          this.renderWelcomeText() : this.renderProtocolList(recentProtocols)
        }
      </div>
    );
  }
}

const mapStateToProps = state => ({
  recentProtocols: get(state, 'recentProtocols', []).slice(0, 4),
});

export { RecentProtocols };

export default connect(mapStateToProps)(RecentProtocols);

