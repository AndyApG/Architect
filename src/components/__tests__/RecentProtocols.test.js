/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { RecentProtocols } from '../RecentProtocols';

const mockProps = {
  recentProtocols: [],
};

describe('<RecentProtocols />', () => {
  it('can render?', () => {
    const component = shallow(<RecentProtocols {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
