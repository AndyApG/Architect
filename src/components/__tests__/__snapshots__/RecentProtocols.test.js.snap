// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentProtocols /> can render? 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentProtocols
    recentProtocols={Array []}
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="recent-protocols__welcome"
      >
        <h1>
          Getting Started
        </h1>
        <p>
          Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.
        </p>
      </div>,
      "className": "recent-protocols",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Getting Started
          </h1>,
          <p>
            Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.
          </p>,
        ],
        "className": "recent-protocols__welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Getting Started",
          },
          "ref": null,
          "rendered": "Getting Started",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.",
          },
          "ref": null,
          "rendered": "Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="recent-protocols__welcome"
        >
          <h1>
            Getting Started
          </h1>
          <p>
            Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.
          </p>
        </div>,
        "className": "recent-protocols",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Getting Started
            </h1>,
            <p>
              Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.
            </p>,
          ],
          "className": "recent-protocols__welcome",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Getting Started",
            },
            "ref": null,
            "rendered": "Getting Started",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.",
            },
            "ref": null,
            "rendered": "Welcome to Network Canvas Architect! To get started, use the buttons above to create a new interview protocol, or open an existing one from elsewhere. When you return to this screen later, recent protocols you have opened will be shown here.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
